How to compile Sandstorm?

@kentonv 

How to compile Sandstorm?

```
$ cat /etc/issue
Debian GNU/Linux buster/sid \n \l

$ uname -a
Linux pc 4.14.0-3-amd64 #1 SMP Debian 4.14.13-1 (2018-01-14) x86_64 GNU/Linux

$ go version
go version go1.9.4 linux/amd64

$ cat build-sandstorm.sh
cd ~/repos/sandstorm/deps/boringssl
rm -rf build
mkdir build
cd build
cmake -GNinja ..
ninja
cd ~/repos/sandstorm
make continuous

$ ./build-sandstorm.sh 
-- The C compiler identification is GNU 7.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- The CXX compiler identification is GNU 7.2.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Perl: /usr/bin/perl (found version "5.26.1") 
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Configuring done
-- Generating done
-- Build files have been written to: /home/lauri/repos/sandstorm/deps/boringssl/build
[395/395] Linking CXX executable decrepit/decrepit_test
cd deps/boringssl/build && cmake .. -DCMAKE_BUILD_TYPE=Release
-- Configuring done
-- Generating done
-- Build files have been written to: /home/lauri/repos/sandstorm/deps/boringssl/build
cd deps/boringssl/build && make -j2
make[1]: Entering directory '/home/lauri/repos/sandstorm/deps/boringssl/build'
make[1]: *** Kohteita ei ole annettu, eikä makefileä löytynyt. Seis.
make[1]: Leaving directory '/home/lauri/repos/sandstorm/deps/boringssl/build'
Makefile:261: recipe for target 'deps/boringssl/build/ssl/libssl.a' failed
make: *** [deps/boringssl/build/ssl/libssl.a] Error 2
```

Quoting Lauri Ojansivu (2018-02-14 11:51:35)

> $ go version
> go version go1.9.4 linux/amd64

This doesn't seem relevant; there's no Go in the sandstorm codebase.

> cd ~/repos/sandstorm/deps/boringssl
> rm -rf build
> mkdir build
> cd build
> cmake -GNinja ..
> ninja

None of this should be necessary; may I ask (1) why you're doing the
above, and (2) what happens without it?

> make continuous
> ...
> make[1]: *** Kohteita ei ole annettu, eikä makefileä löytynyt. Seis.

Could you re-run this as `LC_ALL=C make continuous`? This will make it
easier for native English speakers to troubleshoot.

@zenhack 

Yes there is go, boringssl requires it.

make continuous is what @kentonv said he has only used.

There is no differenct with make, same error.

```
$ make
make[1]: Entering directory '/home/lauri/repos/sandstorm/deps/ekam'
make[1]: 'bin/ekam-bootstrap' is up to date.
make[1]: Leaving directory '/home/lauri/repos/sandstorm/deps/ekam'
cd deps/boringssl/build && cmake .. -DCMAKE_BUILD_TYPE=Release
-- Configuring done
-- Generating done
-- Build files have been written to: /home/lauri/repos/sandstorm/deps/boringssl/build
cd deps/boringssl/build && make -j2
make[1]: Entering directory '/home/lauri/repos/sandstorm/deps/boringssl/build'
make[1]: *** No targets specified and no makefile found.  Stop.
make[1]: Leaving directory '/home/lauri/repos/sandstorm/deps/boringssl/build'
Makefile:261: recipe for target 'deps/boringssl/build/ssl/libssl.a' failed
make: *** [deps/boringssl/build/ssl/libssl.a] Error 2
```

Ok now I get these errors:

```
  compile: libseccomp/api.c
  compile: libseccomp/arch-parisc.c
  compile: libseccomp/arch-ppc64.c
  compile: libseccomp/arch-s390x.c
  compile: libseccomp/arch-ppc.c
  compile: sodium/crypto_core/hsalsa20/core_hsalsa20.c
  compile: sodium/crypto_core/hsalsa20/ref2/core_hsalsa20_ref2.c
  compile: sodium/crypto_core/hchacha20/core_hchacha20.c
  compile: sodium/crypto_core/ed25519/core_ed25519.c
    In file included from /ekam-provider/canonical/sodium/crypto_core/ed25519/core_ed25519.c:4:
    In file included from /ekam-provider/canonical/sodium/private/ed25519_ref10.h:25:
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:79:1: warning: unused function 'fe25519_sub' [-
      Wunused-function]
    fe25519_sub(fe25519 h, const fe25519 f, const fe25519 g)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:148:1: warning: unused function 'fe25519_cmov' [-
      Wunused-function]
    fe25519_cmov(fe25519 f, const fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:198:1: warning: unused function 'fe25519_cswap' [-
      Wunused-function]
    fe25519_cswap(fe25519 f, fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:368:1: warning: unused function 'fe25519_mul' [-
      Wunused-function]
    fe25519_mul(fe25519 h, const fe25519 f, const fe25519 g)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:638:1: warning: unused function 'fe25519_sq' [-
      Wunused-function]
    fe25519_sq(fe25519 h, const fe25519 f)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:810:1: warning: unused function 'fe25519_sq2' [-
      Wunused-function]
    fe25519_sq2(fe25519 h, const fe25519 f)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:982:1: warning: unused function 'fe25519_scalar_
      product' [-Wunused-function]
    fe25519_scalar_product(fe25519 h, const fe25519 f, uint32_t n)
    ^
    ...(log truncated; use -l to increase log limit)...
  compile: sodium/crypto_core/salsa/ref/core_salsa_ref.c
  compile: sodium/randombytes.c
  compile: sodium/crypto_hash.c
  compile: sodium/salsa20/xmm6/salsa20_xmm6.c
  compile: sodium/salsa20/ref/salsa20_ref.c
  compile: sodium/crypto_core/ed25519/ref10/ed25519_ref10.c
    In file included from /ekam-provider/canonical/sodium/crypto_core/ed25519/ref10/ed25519_ref10.c:8:
    In file included from /ekam-provider/canonical/sodium/private/ed25519_ref10.h:25:
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:198:1: warning: unused function 'fe25519_cswap' [-
      Wunused-function]
    fe25519_cswap(fe25519 f, fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:982:1: warning: unused function 'fe25519_scalar_
      product' [-Wunused-function]
    fe25519_scalar_product(fe25519 h, const fe25519 f, uint32_t n)
    ^
    /ekam-provider/canonical/sodium/crypto_core/ed25519/ref10/ed25519_ref10.c:448:1: warning: unused function 'ge25519_
      p3_to_precomp' [-Wunused-function]
    ge25519_p3_to_precomp(ge25519_precomp *pi, const ge25519_p3 *p)
    ^
    3 warnings generated.
  compile: sodium/salsa20/xmm6int/salsa20_xmm6int-avx2.c
  compile: sodium/salsa20/xmm6int/salsa20_xmm6int-sse2.c
  compile: sodium/salsa20/stream_salsa20.c
  compile: sodium/utils.c
    /ekam-provider/canonical/sodium/utils.c:84:22: warning: unused variable 'page_size' [-Wunused-variable]
    static size_t        page_size;
                         ^
    1 warning generated.
  compile: sodium/crypto_pwhash/crypto_pwhash.c
  compile: sodium/randombytes_sysrandom.c
  compile: sodium/crypto_pwhash/scryptsalsa208sha256/sse/pwhash_scryptsalsa208sha256_sse.c
  compile: sodium/crypto_pwhash/scryptsalsa208sha256/scrypt_platform.c
  compile: sodium/crypto_pwhash/scryptsalsa208sha256/crypto_scrypt-common.c
  compile: sodium/crypto_pwhash/scryptsalsa208sha256/nosse/pwhash_scryptsalsa208sha256_nosse.c
  compile: sodium/crypto_pwhash/argon2/blake2b-long.c
  compile: sodium/crypto_pwhash/scryptsalsa208sha256/pwhash_scryptsalsa208sha256.c
  compile: sodium/crypto_pwhash/scryptsalsa208sha256/pbkdf2-sha256.c
  compile: sodium/crypto_pwhash/argon2/argon2-encoding.c
    In file included from /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-encoding.c:2:
    /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-core.h:139:17: warning: unused function 'index_alpha' [-
      Wunused-function]
    static uint32_t index_alpha(const argon2_instance_t *instance,
                    ^
    1 warning generated.
  compile: sodium/crypto_pwhash/argon2/argon2-fill-block-ssse3.c
    In file included from /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-fill-block-ssse3.c:18:
    /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-core.h:139:17: warning: unused function 'index_alpha' [-
      Wunused-function]
    static uint32_t index_alpha(const argon2_instance_t *instance,
                    ^
    1 warning generated.
  compile: sodium/crypto_pwhash/argon2/argon2-fill-block-avx2.c
    In file included from /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-fill-block-avx2.c:18:
    /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-core.h:139:17: warning: unused function 'index_alpha' [-
      Wunused-function]
    static uint32_t index_alpha(const argon2_instance_t *instance,
                    ^
    1 warning generated.
  compile: sodium/crypto_pwhash/argon2/argon2-fill-block-ref.c
    /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-fill-block-ref.c:198:9: warning: unknown pragma ignored 
      [-Wunknown-pragmas]
    #pragma warning(push)
            ^
    /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-fill-block-ref.c:199:9: warning: unknown pragma ignored 
      [-Wunknown-pragmas]
    #pragma warning(disable : 6385)
            ^
    /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-fill-block-ref.c:201:9: warning: unknown pragma ignored 
      [-Wunknown-pragmas]
    #pragma warning(pop)
            ^
    3 warnings generated.
  compile: sodium/crypto_pwhash/argon2/argon2.c
    In file included from /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2.c:22:
    /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-core.h:139:17: warning: unused function 'index_alpha' [-
      Wunused-function]
    static uint32_t index_alpha(const argon2_instance_t *instance,
                    ^
    1 warning generated.
  compile: sodium/crypto_pwhash/argon2/pwhash_argon2id.c
    In file included from /ekam-provider/canonical/sodium/crypto_pwhash/argon2/pwhash_argon2id.c:8:
    /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-core.h:139:17: warning: unused function 'index_alpha' [-
      Wunused-function]
    static uint32_t index_alpha(const argon2_instance_t *instance,
                    ^
    1 warning generated.
  compile: sodium/crypto_pwhash/argon2/argon2-fill-block-avx512f.c
    In file included from /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-fill-block-avx512f.c:18:
    /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-core.h:139:17: warning: unused function 'index_alpha' [-
      Wunused-function]
    static uint32_t index_alpha(const argon2_instance_t *instance,
                    ^
    1 warning generated.
  compile: sodium/crypto_pwhash/argon2/pwhash_argon2i.c
    In file included from /ekam-provider/canonical/sodium/crypto_pwhash/argon2/pwhash_argon2i.c:9:
    /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-core.h:139:17: warning: unused function 'index_alpha' [-
      Wunused-function]
    static uint32_t index_alpha(const argon2_instance_t *instance,
                    ^
    1 warning generated.
  compile: sodium/crypto_pwhash/argon2/argon2-core.c
    In file included from /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-core.c:31:
    /ekam-provider/canonical/sodium/crypto_pwhash/argon2/argon2-core.h:139:17: warning: unused function 'index_alpha' [-
      Wunused-function]
    static uint32_t index_alpha(const argon2_instance_t *instance,
                    ^
    1 warning generated.
  compile: sodium/crypto_stream_chacha20/stream_chacha20.c
  compile: sodium/crypto_stream_chacha20/ref/chacha20_ref.c
  compile: sodium/crypto_stream_chacha20/dolbeau/chacha20_dolbeau-avx2.c
  compile: sodium/crypto_onetimeauth_poly1305/sse2/poly1305_sse2.c
  compile: sodium/crypto_stream_chacha20/dolbeau/chacha20_dolbeau-ssse3.c
  compile: sodium/core.c
    /ekam-provider/canonical/sodium/core.c:34:3: warning: *** The library is being compiled using an undocumented method
      . [-W#warnings]
    # warning *** The library is being compiled using an undocumented method.
      ^
    /ekam-provider/canonical/sodium/core.c:35:3: warning: This is not supported. It has not been tested, it might not [-
      W#warnings]
    # warning This is not supported. It has not been tested, it might not
      ^
    /ekam-provider/canonical/sodium/core.c:36:3: warning: work as expected, and performance is likely to be suboptimal. 
      [-W#warnings]
    # warning work as expected, and performance is likely to be suboptimal.
      ^
    /ekam-provider/canonical/sodium/core.c:40:21: warning: unused variable 'locked' [-Wunused-variable]
    static volatile int locked;
                        ^
    4 warnings generated.
  compile: sodium/crypto_verify/sodium/verify.c
  compile: sodium/crypto_onetimeauth_poly1305/donna/poly1305_donna.c
  compile: sodium/crypto_onetimeauth_poly1305/onetimeauth_poly1305.c
  compile: sodium/crypto_sign/crypto_sign.c
  compile: sodium/crypto_sign/ed25519/ref10/sign.c
    In file included from /ekam-provider/canonical/sodium/crypto_sign/ed25519/ref10/sign.c:7:
    In file included from /ekam-provider/canonical/sodium/private/ed25519_ref10.h:25:
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:79:1: warning: unused function 'fe25519_sub' [-
      Wunused-function]
    fe25519_sub(fe25519 h, const fe25519 f, const fe25519 g)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:148:1: warning: unused function 'fe25519_cmov' [-
      Wunused-function]
    fe25519_cmov(fe25519 f, const fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:198:1: warning: unused function 'fe25519_cswap' [-
      Wunused-function]
    fe25519_cswap(fe25519 f, fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:368:1: warning: unused function 'fe25519_mul' [-
      Wunused-function]
    fe25519_mul(fe25519 h, const fe25519 f, const fe25519 g)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:638:1: warning: unused function 'fe25519_sq' [-
      Wunused-function]
    fe25519_sq(fe25519 h, const fe25519 f)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:810:1: warning: unused function 'fe25519_sq2' [-
      Wunused-function]
    fe25519_sq2(fe25519 h, const fe25519 f)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:982:1: warning: unused function 'fe25519_scalar_
      product' [-Wunused-function]
    fe25519_scalar_product(fe25519 h, const fe25519 f, uint32_t n)
    ^
    ...(log truncated; use -l to increase log limit)...
  compile: sodium/crypto_sign/ed25519/ref10/open.c
    In file included from /ekam-provider/canonical/sodium/crypto_sign/ed25519/ref10/open.c:10:
    In file included from /ekam-provider/canonical/sodium/private/ed25519_ref10.h:25:
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:79:1: warning: unused function 'fe25519_sub' [-
      Wunused-function]
    fe25519_sub(fe25519 h, const fe25519 f, const fe25519 g)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:148:1: warning: unused function 'fe25519_cmov' [-
      Wunused-function]
    fe25519_cmov(fe25519 f, const fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:198:1: warning: unused function 'fe25519_cswap' [-
      Wunused-function]
    fe25519_cswap(fe25519 f, fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:368:1: warning: unused function 'fe25519_mul' [-
      Wunused-function]
    fe25519_mul(fe25519 h, const fe25519 f, const fe25519 g)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:638:1: warning: unused function 'fe25519_sq' [-
      Wunused-function]
    fe25519_sq(fe25519 h, const fe25519 f)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:810:1: warning: unused function 'fe25519_sq2' [-
      Wunused-function]
    fe25519_sq2(fe25519 h, const fe25519 f)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:982:1: warning: unused function 'fe25519_scalar_
      product' [-Wunused-function]
    fe25519_scalar_product(fe25519 h, const fe25519 f, uint32_t n)
    ^
    ...(log truncated; use -l to increase log limit)...
  compile: sodium/crypto_sign/ed25519/ref10/keypair.c
    In file included from /ekam-provider/canonical/sodium/crypto_sign/ed25519/ref10/keypair.c:8:
    In file included from /ekam-provider/canonical/sodium/private/ed25519_ref10.h:25:
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:148:1: warning: unused function 'fe25519_cmov' [-
      Wunused-function]
    fe25519_cmov(fe25519 f, const fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:198:1: warning: unused function 'fe25519_cswap' [-
      Wunused-function]
    fe25519_cswap(fe25519 f, fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:638:1: warning: unused function 'fe25519_sq' [-
      Wunused-function]
    fe25519_sq(fe25519 h, const fe25519 f)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:810:1: warning: unused function 'fe25519_sq2' [-
      Wunused-function]
    fe25519_sq2(fe25519 h, const fe25519 f)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:982:1: warning: unused function 'fe25519_scalar_
      product' [-Wunused-function]
    fe25519_scalar_product(fe25519 h, const fe25519 f, uint32_t n)
    ^
    5 warnings generated.
  compile: sodium/crypto_sign/ed25519/ref10/obsolete.c
    In file included from /ekam-provider/canonical/sodium/crypto_sign/ed25519/ref10/obsolete.c:9:
    In file included from /ekam-provider/canonical/sodium/private/ed25519_ref10.h:25:
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:79:1: warning: unused function 'fe25519_sub' [-
      Wunused-function]
    fe25519_sub(fe25519 h, const fe25519 f, const fe25519 g)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:148:1: warning: unused function 'fe25519_cmov' [-
      Wunused-function]
    fe25519_cmov(fe25519 f, const fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:198:1: warning: unused function 'fe25519_cswap' [-
      Wunused-function]
    fe25519_cswap(fe25519 f, fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:368:1: warning: unused function 'fe25519_mul' [-
      Wunused-function]
    fe25519_mul(fe25519 h, const fe25519 f, const fe25519 g)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:638:1: warning: unused function 'fe25519_sq' [-
      Wunused-function]
    fe25519_sq(fe25519 h, const fe25519 f)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:810:1: warning: unused function 'fe25519_sq2' [-
      Wunused-function]
    fe25519_sq2(fe25519 h, const fe25519 f)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:982:1: warning: unused function 'fe25519_scalar_
      product' [-Wunused-function]
    fe25519_scalar_product(fe25519 h, const fe25519 f, uint32_t n)
    ^
    ...(log truncated; use -l to increase log limit)...
  compile: sodium/crypto_sign/ed25519/sign_ed25519.c
  compile: sodium/crypto_generichash_blake2b/ref/generichash_blake2b.c
  compile: sodium/crypto_generichash_blake2b/ref/blake2b-compress-sse41.c
  compile: sodium/crypto_generichash_blake2b/ref/blake2b-compress-avx2.c
  compile: sodium/crypto_generichash_blake2b/ref/blake2b-compress-ref.c
  compile: sodium/crypto_generichash_blake2b/ref/blake2b-ref.c
  compile: sodium/crypto_generichash_blake2b/ref/blake2b-compress-ssse3.c
  compile: sodium/crypto_generichash_blake2b/generichash_blake2.c
  compile: sodium/crypto_scalarmult/crypto_scalarmult.c
  compile: sodium/crypto_scalarmult/curve25519/sandy2x/curve25519_sandy2x.c
  compile: sodium/crypto_scalarmult/curve25519/ref10/x25519_ref10.c
    In file included from /ekam-provider/canonical/sodium/crypto_scalarmult/curve25519/ref10/x25519_ref10.c:7:
    In file included from /ekam-provider/canonical/sodium/private/ed25519_ref10.h:25:
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:148:1: warning: unused function 'fe25519_cmov' [-
      Wunused-function]
    fe25519_cmov(fe25519 f, const fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:810:1: warning: unused function 'fe25519_sq2' [-
      Wunused-function]
    fe25519_sq2(fe25519 h, const fe25519 f)
    ^
    2 warnings generated.
  compile: sodium/crypto_scalarmult/curve25519/sandy2x/fe51_invert.c
  compile: sodium/crypto_scalarmult/curve25519/sandy2x/fe_frombytes_sandy2x.c
  compile: sodium/crypto_scalarmult/curve25519/scalarmult_curve25519.c
  compile: sodium/crypto_scalarmult/ed25519/ref10/scalarmult_ed25519_ref10.c
    In file included from /ekam-provider/canonical/sodium/crypto_scalarmult/ed25519/ref10/scalarmult_ed25519_ref10.c:5:
    In file included from /ekam-provider/canonical/sodium/private/ed25519_ref10.h:25:
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:79:1: warning: unused function 'fe25519_sub' [-
      Wunused-function]
    fe25519_sub(fe25519 h, const fe25519 f, const fe25519 g)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:148:1: warning: unused function 'fe25519_cmov' [-
      Wunused-function]
    fe25519_cmov(fe25519 f, const fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:198:1: warning: unused function 'fe25519_cswap' [-
      Wunused-function]
    fe25519_cswap(fe25519 f, fe25519 g, unsigned int b)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:368:1: warning: unused function 'fe25519_mul' [-
      Wunused-function]
    fe25519_mul(fe25519 h, const fe25519 f, const fe25519 g)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:638:1: warning: unused function 'fe25519_sq' [-
      Wunused-function]
    fe25519_sq(fe25519 h, const fe25519 f)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:810:1: warning: unused function 'fe25519_sq2' [-
      Wunused-function]
    fe25519_sq2(fe25519 h, const fe25519 f)
    ^
    /ekam-provider/canonical/sodium/private/ed25519_ref10_fe_25_5.h:982:1: warning: unused function 'fe25519_scalar_
      product' [-Wunused-function]
    fe25519_scalar_product(fe25519 h, const fe25519 f, uint32_t n)
    ^
    ...(log truncated; use -l to increase log limit)...
  compile: sodium/crypto_hash_sha256/hash_sha256.c
  compile: sodium/runtime.c
  compile: sodium/crypto_hash_sha256/cp/hash_sha256_cp.c
  compile: sodium/crypto_hash_sha512/hash_sha512.c
  compile: sodium/crypto_hash_sha512/cp/hash_sha512_cp.c
✘ link: kj/test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: capnp/endian-fallback-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: capnp/endian-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: capnp/compiler/lexer-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/common-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/test-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/threadlocal-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/compat/http-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/compat/tls-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/tuple-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/one-of-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/filesystem-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/function-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/parse/char-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/units-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/thread-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/async-unix-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/debug-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/filesystem-disk-old-kernel-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: capnp/compiler/capnpc-c++.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/filesystem-disk-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: capnp/compiler/evolution-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: node-capnp/capnp.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ capnpc: sandstorm/powerbox.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/appid-replacements.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/ip.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/persistentuiview.capnp
    error:  couldn't find capnp.
✘ link: kj/async-io-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ capnpc: sandstorm/app-index/submit.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/mime.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/update-tool.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/package.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/web-session.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/test-app/test-app.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/email-impl.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/appid-replacements-test.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/activity.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/api-session-impl.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/web-publishing.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/grain.capnp
    error:  couldn't find capnp.
✘ capnpc: node-capnp/test.capnp
    error:  couldn't find capnp.
✘ capnpc: capnp/test.capnp
    error:  couldn't find capnp.
✘ capnpc: capnp/test-import2.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/email.capnp
    error:  couldn't find capnp.
✘ link: kj/string-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/parse/common-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: libseccomp/arch-syscall-check.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/exception-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ compile: sandstorm/run-bundle.c++
    /ekam-provider/canonical/sandstorm/run-bundle.c++:27:10: fatal error: 'sandstorm/package.capnp.h' file not found
    #include <sandstorm/package.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/util-test.c++
    In file included from /ekam-provider/canonical/sandstorm/util-test.c++:17:
    /ekam-provider/canonical/sandstorm/util.h:34:10: fatal error: 'sandstorm/util.capnp.h' file not found
    #include <sandstorm/util.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/id-to-text-test.c++
    In file included from /ekam-provider/canonical/sandstorm/id-to-text-test.c++:17:
    /ekam-provider/canonical/sandstorm/id-to-text.h:21:10: fatal error: 'sandstorm/package.capnp.h' file not found
    #include <sandstorm/package.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/backup.c++
    In file included from /ekam-provider/canonical/sandstorm/backup.c++:18:
    /ekam-provider/canonical/sandstorm/util.h:34:10: fatal error: 'sandstorm/util.capnp.h' file not found
    #include <sandstorm/util.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/update-tool.c++
    /ekam-provider/canonical/sandstorm/update-tool.c++:18:10: fatal error: 'sandstorm/update-tool.capnp.h' file not 
      found
    #include <sandstorm/update-tool.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/sandstorm-http-bridge.c++
    /ekam-provider/canonical/sandstorm/sandstorm-http-bridge.c++:47:10: fatal error: 'sandstorm/util.capnp.h' file not 
      found
    #include <sandstorm/util.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/bridge-proxy.c++
    In file included from /ekam-provider/canonical/sandstorm/bridge-proxy.c++:17:
    /ekam-provider/canonical/sandstorm/bridge-proxy.h:21:10: fatal error: 'sandstorm/sandstorm-http-bridge.capnp.h' file
      not found
    #include <sandstorm/sandstorm-http-bridge.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/app-index/app-index.c++
    /ekam-provider/canonical/sandstorm/app-index/app-index.c++:38:10: fatal error: 'sandstorm/grain.capnp.h' file not 
      found
    #include <sandstorm/grain.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/util.c++
    In file included from /ekam-provider/canonical/sandstorm/util.c++:17:
    /ekam-provider/canonical/sandstorm/util.h:34:10: fatal error: 'sandstorm/util.capnp.h' file not found
    #include <sandstorm/util.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/test-app/test-app.c++
    /ekam-provider/canonical/sandstorm/test-app/test-app.c++:29:10: fatal error: 'sandstorm/grain.capnp.h' file not 
      found
    #include <sandstorm/grain.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/backend.c++
    In file included from /ekam-provider/canonical/sandstorm/backend.c++:17:
    /ekam-provider/canonical/sandstorm/backend.h:20:10: fatal error: 'sandstorm/backend.capnp.h' file not found
    #include <sandstorm/backend.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/id-to-text.c++
    In file included from /ekam-provider/canonical/sandstorm/id-to-text.c++:17:
    /ekam-provider/canonical/sandstorm/id-to-text.h:21:10: fatal error: 'sandstorm/package.capnp.h' file not found
    #include <sandstorm/package.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/fuse.c++
    In file included from /ekam-provider/canonical/sandstorm/fuse.c++:19:
    /ekam-provider/canonical/sandstorm/util.h:34:10: fatal error: 'sandstorm/util.capnp.h' file not found
    #include <sandstorm/util.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/appid-replacements-test.c++
    In file included from /ekam-provider/canonical/sandstorm/appid-replacements-test.c++:17:
    /ekam-provider/canonical/sandstorm/appid-replacements.h:20:10: fatal error: 'sandstorm/appid-replacements.capnp.h' 
      file not found
    #include <sandstorm/appid-replacements.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ link: kj/io-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ compile: sandstorm/spk.c++
    In file included from /ekam-provider/canonical/sandstorm/spk.c++:19:
    /ekam-provider/canonical/sandstorm/spk.h:23:10: fatal error: 'sandstorm/package.capnp.h' file not found
    #include <sandstorm/package.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ link: sandstorm/fuse-test-bind.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ compile: sandstorm/gateway.c++
    In file included from /ekam-provider/canonical/sandstorm/gateway.c++:17:
    /ekam-provider/canonical/sandstorm/gateway.h:21:10: fatal error: 'sandstorm/backend.capnp.h' file not found
    #include <sandstorm/backend.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/supervisor.c++
    In file included from /ekam-provider/canonical/sandstorm/supervisor.c++:17:
    In file included from /ekam-provider/canonical/sandstorm/supervisor.h:21:
    /ekam-provider/canonical/sandstorm/util.h:34:10: fatal error: 'sandstorm/util.capnp.h' file not found
    #include <sandstorm/util.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: capnp/serialize-async-test.c++
    In file included from /ekam-provider/canonical/capnp/serialize-async-test.c++:28:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: capnp/compat/json-test.c++
    In file included from /ekam-provider/canonical/capnp/compat/json-test.c++:23:
    /ekam-provider/c++header/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ capnpc: sandstorm/bridge-proxy.capnp
    error:  couldn't find capnp.
✘ compile: capnp/dynamic-test.c++
    In file included from /ekam-provider/canonical/capnp/dynamic-test.c++:26:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ capnpc: sandstorm/util.capnp
    error:  couldn't find capnp.
✘ capnpc: sandstorm/backend.capnp
    error:  couldn't find capnp.
✘ link: kj/filesystem-disk-generic-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: capnp/compiler/capnpc-capnp.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: libseccomp/arch-syscall-dump.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ capnpc: sandstorm/api-session.capnp
    error:  couldn't find capnp.
✘ compile: sandstorm/app-index/indexer.c++
    In file included from /ekam-provider/canonical/sandstorm/app-index/indexer.c++:17:
    /ekam-provider/canonical/sandstorm/app-index/indexer.h:21:10: fatal error: 'sandstorm/util.capnp.h' file not found
    #include <sandstorm/util.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ link: kj/memory-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ capnpc: sandstorm/identity.capnp
    error:  couldn't find capnp.
✘ compile: capnp/schema-test.c++
    In file included from /ekam-provider/canonical/capnp/schema-test.c++:26:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/web-session-bridge.c++
    In file included from /ekam-provider/canonical/sandstorm/web-session-bridge.c++:17:
    /ekam-provider/canonical/sandstorm/web-session-bridge.h:21:10: fatal error: 'sandstorm/web-session.capnp.h' file not
      found
    #include <sandstorm/web-session.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: capnp/serialize-test.c++
    In file included from /ekam-provider/canonical/capnp/serialize-test.c++:29:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ learn: capnp/bootstrap-test.ekam-rule
    error:  couldn't find capnpc-c++.
✘ compile: capnp/rpc-twoparty-test.c++
    In file included from /ekam-provider/canonical/capnp/rpc-twoparty-test.c++:25:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ link: kj/string-tree-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ compile: capnp/encoding-test.c++
    /ekam-provider/canonical/capnp/encoding-test.c++:22:10: fatal error: 'capnp/test-import.capnp.h' file not found
    #include <capnp/test-import.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ capnpc: capnp/test-import.capnp
    error:  couldn't find capnp.
✘ compile: capnp/common-test.c++
    /ekam-provider/canonical/capnp/common-test.c++:26:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ link: kj/encoding-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ compile: sandstorm/smtp-proxy.c++
    In file included from /ekam-provider/canonical/sandstorm/smtp-proxy.c++:17:
    In file included from /ekam-provider/canonical/sandstorm/smtp-proxy.h:19:
    /ekam-provider/canonical/sandstorm/util.h:34:10: fatal error: 'sandstorm/util.capnp.h' file not found
    #include <sandstorm/util.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: capnp/test-util.c++
    In file included from /ekam-provider/canonical/capnp/test-util.c++:22:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ capnpc: sandstorm/app-index/app-index.capnp
    error:  couldn't find capnp.
✘ compile: sandstorm/union-fs.c++
    In file included from /ekam-provider/canonical/sandstorm/union-fs.c++:17:
    /ekam-provider/canonical/sandstorm/union-fs.h:23:10: fatal error: 'sandstorm/package.capnp.h' file not found
    #include <sandstorm/package.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ capnpc: sandstorm/identity-impl.capnp
    error:  couldn't find capnp.
✘ compile: capnp/rpc-test.c++
    In file included from /ekam-provider/canonical/capnp/rpc-test.c++:25:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ install: node-capnp/node-capnp.ekam-manifest
    node-capnp/capnp.node: not found
✘ install: capnp/compiler/capnp.ekam-manifest
    capnp/compiler/capnp: not found
✘ capnpc: sandstorm/sandstorm-http-bridge.capnp
    error:  couldn't find capnp.
✘ link: capnp/endian-reverse-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ compile: capnp/schema-loader-test.c++
    In file included from /ekam-provider/canonical/capnp/schema-loader-test.c++:26:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ link: kj/arena-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ compile: capnp/stringify-test.c++
    In file included from /ekam-provider/canonical/capnp/stringify-test.c++:27:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ link: capnp/compiler/capnp.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ install: sandstorm/sandstorm.ekam-manifest
    sandstorm/run-bundle: not found
✘ capnpc: sandstorm/supervisor.capnp
    error:  couldn't find capnp.
✘ link: kj/mutex-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/compat/url-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ link: kj/compat/readiness-io-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ learn: capnp/compiler/capnp-test.ekam-rule
    src/capnp/compiler/capnp-test.sh: 53: src/capnp/compiler/capnp-test.sh: src/capnp: Permission denied
    cmp: EOF on - which is empty
    FAILED: encode
✘ link: kj/refcount-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ capnpc: sandstorm/sandstorm-http-bridge-internal.capnp
    error:  couldn't find capnp.
✘ link: kj/array-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ compile: capnp/serialize-packed-test.c++
    In file included from /ekam-provider/canonical/capnp/serialize-packed-test.c++:27:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: capnp/afl-testcase.c++
    In file included from /ekam-provider/canonical/capnp/afl-testcase.c++:22:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/appid-replacements.c++
    In file included from /ekam-provider/canonical/sandstorm/appid-replacements.c++:17:
    /ekam-provider/canonical/sandstorm/appid-replacements.h:20:10: fatal error: 'sandstorm/appid-replacements.capnp.h' 
      file not found
    #include <sandstorm/appid-replacements.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: capnp/blob-test.c++
    In file included from /ekam-provider/canonical/capnp/blob-test.c++:26:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: sandstorm/spk-standalone.c++
    In file included from /ekam-provider/canonical/sandstorm/spk-standalone.c++:17:
    /ekam-provider/canonical/sandstorm/spk.h:23:10: fatal error: 'sandstorm/package.capnp.h' file not found
    #include <sandstorm/package.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: capnp/serialize-text-test.c++
    In file included from /ekam-provider/canonical/capnp/serialize-text-test.c++:27:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: capnp/canonicalize-test.c++
    In file included from /ekam-provider/canonical/capnp/canonicalize-test.c++:26:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ link: capnp/compiler/type-id-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ compile: capnp/schema-parser-test.c++
    In file included from /ekam-provider/canonical/capnp/schema-parser-test.c++:26:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: capnp/any-test.c++
    In file included from /ekam-provider/canonical/capnp/any-test.c++:25:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ link: capnp/layout-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ compile: capnp/capability-test.c++
    /ekam-provider/canonical/capnp/capability-test.c++:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: capnp/fuzz-test.c++
    /ekam-provider/canonical/capnp/fuzz-test.c++:22:10: fatal error: 'capnp/test-import.capnp.h' file not found
    #include <capnp/test-import.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: capnp/ez-rpc-test.c++
    In file included from /ekam-provider/canonical/capnp/ez-rpc-test.c++:25:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ link: kj/compat/gzip-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ compile: capnp/message-test.c++
    In file included from /ekam-provider/canonical/capnp/message-test.c++:23:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ compile: capnp/membrane-test.c++
    In file included from /ekam-provider/canonical/capnp/membrane-test.c++:24:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ link: kj/threadlocal-pthread-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ compile: capnp/orphan-test.c++
    In file included from /ekam-provider/canonical/capnp/orphan-test.c++:25:
    /ekam-provider/canonical/capnp/test-util.h:28:10: fatal error: 'capnp/test.capnp.h' file not found
    #include <capnp/test.capnp.h>
             ^~~~~~~~~~~~~~~~~~~~
    1 error generated.
✘ jstest: node-capnp/capnp-test.js
    full log: tmp/node-capnp/capnp-test.js.log
✘ link: kj/async-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
✘ capnpc: sandstorm/hack-session.capnp
    error:  couldn't find capnp.
✘ link: kj/std/iostream-test.o
    /usr/bin/ld: cannot find -lz
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
==== build failed. You might need to: make update-deps ====
Makefile:274: recipe for target 'tmp/.ekam-run' failed
make: *** [tmp/.ekam-run] Error 1
```

--- a/docs/install.md
+++ b/docs/install.md
@@ -138,6 +138,8 @@ Please install the following:
 * `strace`
 * `curl`
 * `python`
+* `zlib1g-dev`
+* `golang-go`
 * `cmake`
 * discount (markdown parser)
 * [Clang compiler](http://clang.llvm.org/) version 3.4 or better
@@ -146,7 +148,8 @@ Please install the following:
 On Debian or Ubuntu, you should be able to get all these with:
 
     sudo apt-get install build-essential libcap-dev xz-utils zip \
-        unzip strace curl clang discount git python
+        unzip strace curl clang discount git python zlib1g-dev \
+        golang-go
     curl https://install.meteor.com/ | sh
 
 ### Get the source code

